netbench-orchestrator/src/russula/protocol.rs

```diff
@@ -0,0 +1,246 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{
+    error::RussulaError,
+    event::EventType,
+    network_utils,
+    network_utils::Msg,
+    states::{StateApi, TransitionStep},
+    ProtocolState, RussulaResult,
+};
+use crate::russula::event::EventRecorder;
+use core::{task::Poll, time::Duration};
+use std::net::SocketAddr;
+use tokio::net::TcpStream;
+use tracing::{debug, info};
+
+// Send the Done status multiple times to the peer in case there is packet loss.
```

Comment:
The goal was to not rely on the transport.. because we might replace TCP with something else. The main purpose of this mechanism is to send the terminal state more than once to the peer since we are shutting down our side and wont be interacting with the peer after this. Besides packet_loss there are other reason in a distributed system why this is necessary (delay)

I was glossing over all this detail but can see why `packet_loss` is a confusing term. I'll remove it and simplify the wording.